xen/arch/arm/coproc/coproc.c

```diff
@@ -353,24 +350,29 @@ struct coproc_device *coproc_alloc(struct platform_device *pdev,
 
     for ( i = 0; i < num_mmios; ++i )
     {
-        res = platform_get_resource(pdev, IORESOURCE_MEM, i);
-        coproc->mmios[i].base = devm_ioremap_resource(dev, res);
+        ret = dt_device_get_address(pdev, i,
+                                    &coproc->mmios[i].addr,
+                                    &coproc->mmios[i].size);
+        if ( ret )
+        {
+            printk(XENLOG_ERR "Unable to retrieve address %u for %s\n",
+                   i, dev_path(coproc->dev));
+            goto out;
```

Comment:
Actually I have changed it to "Failed to get mmio index %d for", similar to your own comment for irqs forty lines below. I'm not happy with both of them, but looking forward to rewrite all that stuff I would not  focus on them.